endoreg_db/models/media/video/video_file_io.py

```
@@ -0,0 +1,130 @@
+        if video.is_processed and video.processed_file.name:
+            return Path(video.processed_file.path)
+        else:
+            return None
+    except Exception as e:
+        logger.warning("Could not get path for processed file of VideoFile %s: %s", video.uuid, e)
+        return None
+
+def _delete_with_file(video: "VideoFile", *args, **kwargs):
+    """Deletes the VideoFile record and its associated physical files (raw, processed, frames)."""
+    logger.warning("Attempting to delete VideoFile record %s (UUID: %s) and associated files.", video.pk, video.uuid)
+    ic(f"Deleting VideoFile {video.uuid} and files.")
```

Ruff (0.8.2)

Undefined name `ic`

(F821)